go/parquet/internal/encoding/boolean_decoder.go

```diff
@@ -109,3 +114,76 @@ func (dec *PlainBooleanDecoder) DecodeSpaced(out []bool, nullCount int, validBit
 	}
 	return dec.Decode(out)
 }
+
+type RleBooleanDecoder struct {
+	decoder
+
+	rleDec *utils.RleDecoder
+}
+
+func (RleBooleanDecoder) Type() parquet.Type {
+	return parquet.Types.Boolean
+}
+
+func (dec *RleBooleanDecoder) SetData(nvals int, data []byte) error {
+	dec.nvals = nvals
+
+	if len(data) < 4 {
+		return fmt.Errorf("invalid length - %d (corrupt data page?)", len(data))
+	}
+
+	// load the first 4 bytes in little-endian which indicates the length
+	nbytes := binary.LittleEndian.Uint32(data[:4])
+	if nbytes > uint32(len(data)-4) {
+		return fmt.Errorf("received invalid number of bytes - %d (corrupt data page?)", nbytes)
+	}
+
+	dec.data = data[4:]
+	if dec.rleDec == nil {
+		dec.rleDec = utils.NewRleDecoder(bytes.NewReader(dec.data), 1)
+	} else {
+		dec.rleDec.Reset(bytes.NewReader(dec.data), 1)
+	}
+	return nil
+}
+
+func (dec *RleBooleanDecoder) Decode(out []bool) (int, error) {
+	max := shared_utils.MinInt(len(out), dec.nvals)
+
+	var (
+		buf [1024]uint64
+		n   = max
```

Comment:
I'm curious, what is the rationale for using `var` vs. `:=`? Is there a well-known guideline for this?

Reply:
In general for me, I tend to use the following guidelines:

1. If I'm declaring a group of variables, i'll put them together into a `var` block to improve readability
2. If i'm not using a block / grouped declaration, then if I'm just using the zero-value I'll use `var`, (like declaring an array / uninitialized slice)
3. If i'm not using a block / grouped declaration, I'll generally use `:=` if I'm initializing it to a particular value. 

Otherwise it's generally a purely aesthetic decision of code style